src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java

```diff
@@ -0,0 +1,215 @@
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_USER_NOT_AUTHENTICATED;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ANSWER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PASSWORD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PASSWORD_CONFIRM;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SECRET_QUESTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_USER;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.user.Password;
+import seedu.address.model.user.User;
+import seedu.address.model.user.Username;
+
+/**
+ * Updates the user's account details.
+ */
+public class UserUpdateCommand extends Command {
+
+    public static final String COMMAND_WORD = "update";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Update your account details.\n"
+            + "Parameters: "
+            + "[" + PREFIX_USER + " USERNAME] "
+            + "[" + PREFIX_PASSWORD + " PASSWORD "
+            + PREFIX_PASSWORD_CONFIRM + " CONFIRM_PASSWORD] "
+            + "[" + PREFIX_SECRET_QUESTION + " SECRET_PASSWORD "
+            + PREFIX_ANSWER + " ANSWER]\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_PASSWORD + " yourNewPassword "
+            + PREFIX_PASSWORD_CONFIRM + " yourNewPassword ";
+    public static final String MESSAGE_SUCCESS = "Update successful.";
+    public static final String MESSAGE_MISSING_FIELDS = "Please provide at least one field to update! \n%1$s";
+    public static final String MESSAGE_PASSWORD_OR_CONFIRM_PASSWORD_MISSING = "Password and Confirm Password "
+            + "have to be either all present or all absent. Try again.";
+    public static final String MESSAGE_PASSWORD_MISMATCH = "Passwords do not match. Please try again.";
+    public static final String MESSAGE_QUESTION_OR_ANSWER_MISSING = "Secret Question and Answer have to be "
+            + "either all present or all absent. Try again.";
+    private final UserUpdateDescriptor userUpdateDescriptor;
+
+    /**
+     * Creates a UserLoginCommand to log in the specified {@code User}
+     */
+    public UserUpdateCommand(UserUpdateDescriptor userUpdateDescriptor) {
+        requireNonNull(userUpdateDescriptor);
+        this.userUpdateDescriptor = userUpdateDescriptor;
+    }
+
+    /**
+     * Executes the register user command.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return {@code CommandResult} that indicates success.
+     * @throws CommandException if the user is already logged in or the user credentials are wrong.
```

Comment:
Are the checks for user credentials being done in this file

Reply:
Good spot, will update the comments accordingly